### DISCUSSION TITLE
Question about link between technique G201 and Criterions

### DISCUSSION THREAD

**Discussion by marie450** [w3c/wcag#3797] [Q&A] [answered]
Can you explain how the Technique "G201: Giving users advanced warning when opening a new window" is link to "3.2.1: On Focus" and "3.2.2: On Input"? The team of accessibility advisors I work with is trying to push for it's use and understanding how it's connected to WCAG A criterion would really help with that even if the link is only at an advisory level.

**Comment by mraccess77** [comment]
It is only advisory for level A and AA.  I believe it is only tangentially related to these success criteria and didn't fit anywhere else but was included because folks at some time thought it was important.  Perhaps at the time there were folks who wanted it to be an A/AA failure - but the consensus fell on it not being a failure and these advisories were added to address their concerns.

**Reply by marie450** [reply]
The problem is that, we can find how it relates to those 2 criteria even at an advisory, or very superficial, level. If we could it would be a great argument for it's addition to our best practices since those criteria are link to a level A that is enforced by law in our contexte. If it didn't fit anywhere it would have been link to a rule, a principle, or nothing instead like other techniques (rares but there).

I fell like G201 would no help users in a change on context on focus as the change would likely happen before the user notice the icon or text. We also believe that it is not applicable to change of context on input if clicking a link is not considered an input per the criterion.

We are really scratching ours heads as to why G201 is an advisory to 3.2.1 and 3.2.2. Maybe it's a mistake or a relic of and old version? We agree that it's great for accessibility to inform users of a link initiating a change of context, sadly Designers don't find it appealing so we need to beef up the case for it.

**Reply by patrickhlauke** [reply]
As written, you're quite right that there isn't a direct connection there, as the new windows aren't opened on focus or on input. I can only imagine that at some point somebody thought "oh, what about cases where a new window is opened on focus, or when changing the value of a form control, or similar?" but didn't actually go through changing the technique itself.

**Reply by kaveinthran** [reply]
I can attest to this, there are instances in some pages where any form manipulation changes the page views or even go to new pages, but it is less nowadays.
are there any ways to view discussions relating to this techniques when it unfolded back then? like hanserd for parliaments or wikipedia talk discussion and page edit history?

**Reply by patrickhlauke** [reply]
this was likely written back in ... 2000 or so. not sure there's much point in trying to excavate that far back. techniques are just examples, they're not normative, and they're far from complete. personally, i don't even bother most of the time even looking at them anymore because they're often so out of date/incomplete...

**Reply by chaals** [reply, marked as answer]
I believe the original discussion of this was in the late 1990s, leading to https://www.w3.org/TR/WAI-WEBCONTENT-TECHS/#tech-avoid-pop-ups for WCAG 1, [checkpoint 10.1](https://www.w3.org/TR/WCAG10/#gl-interim-accessibility). 

There *are* records - email archives, minutes of meetings (very like Hansard!), edit records... 

I've gone through the [change history between WCAG 2.0 drafts](https://www.w3.org/WAI/GL/WCAG20/change-history.html) for the following potted history of what happened n the spec itself. With more time (I spent an hour) you can probably find most of the rationales and discussions in email archives and meeting minutes - there are links scattered around.

Note that techniques, the understanding docs, etc, were developed in parallel - in the early history there were WCAG 1 techniques available, but most stuff including the structures developed over the 7-odd years it took to produce WCAG 2.

In the first draft of WCAG 2 (2001) this became "[2.3 Give users control of mechanisms that cause extreme changes in context](https://www.w3.org/TR/2001/WD-WCAG20-20010125/#extreme-context-change)".

In the April 2004 draft it had changed to [Checkpoint 3.5 [2.2 and 2.3] Provide consistent and predictable responses to user actions.](https://www.w3.org/WAI/GL/WCAG20/WD-WCAG20-20020424.html#consistent-responses)

over the next few years this shuffled a bit and got some L1 (equivalent to level A) success criteria that cover this case.

In June 2005 the success criteria started changing again [3.2 Make the placement and functionality of content predictable](https://www.w3.org/TR/2005/WD-WCAG20-20050630/#consistent-behavior) to require at level 1 that what happened was "programmatically determinable". The discussion around that might be relevant to read.

By the [November 2005 draft](https://www.w3.org/TR/2005/WD-WCAG20-20051123/guidelines.html#consistent-behavior) the success criteria were significantly different, and G201 (based on stuff that had been L1) would only apply to triple-A, which carried through the rest of the WCAG 2 development.

**Reply by marie450** [reply]
Would you say that, now, G201 should be link to 3.2.5 instead?
